SpiceSharp/Sparse/Matrix.cs

```diff
@@ -0,0 +1,426 @@
+                {
+                    // Next one maybe?
+                    elt = elt.NextInCol;
+                }
+                else if (elt.Row == row)
+                {
+                    // Found it!
+                    return elt;
+                }
+                else
+                {
+                    // Aww...
+                    return null;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Create a new element in the matrix if it doesn't exist
+        /// Only used internally!
+        /// </summary>
+        /// <param name="matrix">Matrix</param>
+        /// <param name="row">Row</param>
+        /// <param name="col">Column</param>
+        /// <returns></returns>
+        private MatrixElement CreateElement(int row, int col)
+        {
+            MatrixElement elt = FirstInCol[col], last = null;
+
+            // Splice into the column vector while also searching for an existing element
+            if (elt == null || elt.Row > row)
+            {
+                // There are no elements yet in the column
+                elt = new MatrixElement(row, col);
+                elt.NextInCol = FirstInCol[col];
+                FirstInCol[col] = elt;
+                if (row == col)
+                    Diag[row] = elt;
+
+                if (RowsLinked)
+                    SpliceInRows(elt);
+            }
+            else
+            {
+                // Find the insert point
+                while (elt != null && elt.Row < row)
+                {
+                    last = elt;
+                    elt = elt.NextInCol;
+                }
+
+                // If the element does not exist yet, create it
+                if (elt == null || elt.Row != row)
+                {
+                    elt = new MatrixElement(row, col);
+                    elt.NextInCol = last.NextInCol;
+                    last.NextInCol = elt;
+
+                    if (row == col)
+                        Diag[row] = elt;
+
+                    // Splice it in the row vector
+                    if (RowsLinked)
+                        SpliceInRows(elt);
+                }
+            }
+            return elt;
+        }
+
+        /// <summary>
+        /// Create a fillin matrix element
+        /// </summary>
+        /// <param name="matrix">Matrix</param>
+        /// <param name="Row">Row</param>
+        /// <param name="Col">Column</param>
+        /// <returns></returns>
+        internal MatrixElement CreateFillin(int Row, int Col)
+        {
+            MatrixElement pElement, aboveElement;
+
+            // Find Element above fill-in. 
+            // ppElementAbove = &matrix.FirstInCol[Col];
+            aboveElement = null;
+            pElement = FirstInCol[Col];
+            while (pElement != null)
+            {
+                if (pElement.Row < Row)
+                {
+                    aboveElement = pElement;
```

Comment:
@svenboulanger AboveElement is never used in following code

Reply:
Even worse, `pElement` is overwritten by `CreateElement(Row, Col)` so the whole part is actually unnecessary.

Reply:
Small note: originally `CreateElement` needed an extra parameter for the element just before the insertion point (it is a singly linked list). This was removed here.